### PR TITLE
feat(meta): let `pin_snapshot` be intra-meta method

### PR DESCRIPTION
**Pull request by soundOfDestiny**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Refactor in #5042 makes `pin_version` an intra-meta method.
This PR will make `pin_snapshot` an intra-meta method.

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Review by zwang28 — COMMENTED** [review]
LGTM. PTAL @Little-Wallace @BowenXiao1999 
We can remove pin_snapshot RPC interface.

**Comment by BowenXiao1999** [comment]
So after this PR, FE can not pin snapshot voluntarily?

**Comment by zwang28** [comment]
> So after this PR, FE can not pin snapshot voluntarily?

Yes, because we don't see any usage of FrontendMetaClient::pin_snapshot. Will it be used later?

**Comment by BowenXiao1999** [comment]
> > So after this PR, FE can not pin snapshot voluntarily?
> 
> Yes, because we don't see any usage of FrontendMetaClient::pin_snapshot. Will it be used later?

Actually it depends on whether we need strong consistency. I actually agree with removed it now, given that we already have "flush". When we have such requirement from user, we can add it back easily.

cc @hzxa21

**Review by BowenXiao1999 — APPROVED** [review]

**Comment by zwang28** [comment]
> Actually it depends on whether we need strong consistency.

OK, then we just won't remove RPC in this PR.

File: src/rpc_client/src/meta_client.rs [hunk 1]
```diff
@@ -112,7 +112,6 @@ impl MetaClient {
         self.set_worker_id(worker_node.id);
         // unpin snapshot before MAX will create a new snapshot with last committed epoch and then
         //  we do not create snapshot during every pin_snapshot.
```
Comment (HuaHuaY) [inline src/rpc_client/src/meta_client.rs:114]: Does this comment still make sense?

Comment (soundOfDestiny) [reply src/rpc_client/src/meta_client.rs:114]: > Does this comment still make sense?

the comment means that we are replacing `pin_version` calls after `register` with `unpin_version_before`

**Pull request merged**